### DISCUSSION TITLE
FCM -  Generate a custom token by using getToken() method or smt else?

### DISCUSSION THREAD

**Discussion by burakkbilginn** [invertase/react-native-firebase#7620] [Q&A]
Is there a way to register a device with a custom FCM token which is a spesific token or a previously generated/deleted token? Like messaging().getToken("fdasafasdfasfdsadf") ?
I need this because I need to use the previously created token

**Comment by mikehardy** [comment]
Hi there! No, unfortunately the SDKs have no API to set the token. They will fetch one for you if you ask but it is an internal process

https://firebase.google.com/docs/reference/js/messaging_
